src/main/java/com/google/firebase/remoteconfig/RemoteConfigParameter.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig;
+
+import com.google.firebase.remoteconfig.internal.TemplateResponse.ParameterResponse;
+import com.google.firebase.remoteconfig.internal.TemplateResponse.ParameterValueResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a Remote Config parameter that can be included in a {@link RemoteConfigTemplate}.
+ * At minimum, a default value or a conditional value must be present for the
+ * parameter to have any effect.
+ */
+public final class RemoteConfigParameter {
+
+  private RemoteConfigParameterValue defaultValue;
+  private String description;
+  private Map<String, RemoteConfigParameterValue> conditionalValues;
+
+  /**
+   * Creates a new {@link RemoteConfigParameter}.
+   */
+  public RemoteConfigParameter() {
+    conditionalValues = new HashMap<>();
+    defaultValue = RemoteConfigParameterValue.of("");
+  }
+
+  /**
+   * Gets the default value of the parameter.
+   *
+   * @return A {@link RemoteConfigParameterValue} instance.
+   */
+  public RemoteConfigParameterValue getDefaultValue() {
+    return defaultValue;
+  }
+
+  /**
+   * Gets the description of the parameter.
+   *
+   * @return The {@link String} description of the parameter.
+   */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Gets the conditional values of the parameter.
+   * The condition name of the highest priority (the one listed first in the
```

Comment:
This is interesting . . . you mean this gets all the possible values, including the one actually assigned -- for a particular client instance? -- in the top position in the map?

Reply:
This gets all the conditional values assigned for that specific parameter. You can set a default value in a parameter to be used when all the conditional values are evaluated to false. Otherwise, the parameter will use the conditional value with the highest priority. The priority is determined by the order of the conditions are stored in the top-level `conditions` array of the `RemoteConfigTemplate`. The top-level `conditions` array is not part of this PR and will be added in a future PR.
https://firebase.google.com/docs/reference/remote-config/rest/v1/RemoteConfigParameter

Reply:
OK, thanks for the explanation.  Parameter evaluation is pretty complex . . . when everything is done, I might try to start a review of our "parameters and conditions" concepts page, just to make sure it's all still accurate.